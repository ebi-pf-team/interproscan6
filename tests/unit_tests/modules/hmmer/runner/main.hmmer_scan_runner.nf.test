nextflow_process {

    name "Test Process HMMER_SCAN_RUNNER"
    script "interproscan/modules/hmmer/runner/main.nf"
    process "HMMER_SCAN_RUNNER"

    stage {
        copy "tests/unit_tests/test_inputs/hmmer_runners/hmmer_runner_fasta.faa"
        copy "tests/unit_tests/test_inputs/hmmer_runners/mock_hmm.hmm"
    }

    test("Should run without failures") {

        when {
            params {}
            process {
                """
                input[0] = [
                    file("meta/tests/unit_tests/test_inputs/hmmer_runners/hmmer_runner_fasta.faa"),
                    "unit-test",
                    file("meta/tests/unit_tests/test_inputs/hmmer_runners/mock_hmm.hmm"),
                    ["switch"],
                    []
                ]
                input[1] = true
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
