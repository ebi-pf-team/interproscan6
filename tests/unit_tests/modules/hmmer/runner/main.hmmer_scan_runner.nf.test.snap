{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "unit-test_out:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "1": [
                    [
                        
                    ]
                ],
                "2": [
                    "unit-test"
                ],
                "3": [
                    "hmmer_runner_fasta.faa:md5,f94265dbb12341fa2aea5906115fe896"
                ],
                "4": [
                    "mock_hmm.hmm:md5,c35f5fe34013624a40fd449d56b1da9d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-09-20T17:56:31.652598"
    }
}